### PR TITLE
Converting BLT to composer package.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -54,26 +58,28 @@ script:
   # Ensure code quality of 'blt' itself.
   - phpcs --standard=./vendor/drupal/coder/coder_sniffer/Drupal/ruleset.xml tests
   # Generate a new 'blted' project.
-  - ./blt.sh configure
+  - cd blt-project
+  - composer install
+  - ./vendor/acquia/blt/blt.sh blt:alias
+  - source ~/.bash_profile
```
Comment (kylebrowning) [inline .travis.yml]: This only works with bash!

**Comment by kylebrowning** [comment]
Will there be an upgrade path from projects using the old blt setup?

**Comment by grasmash** [comment]
Yeah I'm planning to have an automated upgrade path.

**Comment by janaksingh** [comment]
I think this build also failed due to issue raised here:
https://github.com/acquia/blt/issues/223

**Comment by ChuChuNaKu** [comment]
I got stuck on step number [4 of the installation instructions](https://github.com/acquia/blt/pull/217/files#diff-7d442b7eb49f5fc377f51e74b291cfc1R58) . I received the following error after running ./vendor/acquia/blt/blt.sh install-alias

> Phing was not found in this project's bin directory.
Attempting to run composer install. This takes a few minutes.
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
./vendor/acquia/blt/blt.sh: line 23: [PROJECT DIRECTORY]/myproject/vendor/acquia/blt/vendor/bin/phing: No such file or directory

**Comment by grasmash** [comment]
@ChuChuNaKu Until this PR has been merged you'll need to use the feature branch for the composer dependency. `composer require acquia/blt:dev-composer-package`

**Comment by pixlkat** [comment]
Running the command ```blt init``` replaces the initial composerjson file created during composer init with the one from the templates directory in blt. This removes the composer requirement for blt.

This is the contents of my composer.json file after Step 4 (require the blt project):
```
{
    "name": "acquia/bltcomposer",
    "description": "BLT composer testing",
    "authors": [
        {
            "name": "Stephanie Bridges",
            "email": "steph.bridges@gmail.com"
        }
    ],
    "minimum-stability": "dev",
    "require": {},
    "prefer-stable": true,
    "require-dev": {
        "acquia/blt": "dev-composer-package"
    }
}
```

After completing Step 5 (initializing BLT), this is my composer.json file:
```
{
    "license": "proprietary",
    "type": "project",
    "repositories": [
        {
            "type": "composer",
            "url": "https://packagist.drupal-composer.org"
        },
        {
            "type": "git",
            "url": "https://github.com/FloeDesignTechnologies/composer-patches"
        }
    ],
    "require": {
        "composer/installers": "^1.0.20",
        "cweagans/composer-patches": "dev-master as 1.5.0",
        "drupal/acquia_connector": "8.1.*",
        "drupal/acsf": "~8.1",
        "drupal/core": "~8",
        "drupal/console": "~1",
        "drupal/memcache": "8.*",
        "drupal/security_review": "8.*",
        "roave/security-advisories": "dev-master",
        "drupal-composer/drupal-security-advisories": "8.0.x-dev",
        "drupal/lightning": "~8.1"
    },
    "require-dev": {
        "behat/behat": "3.0.*",
        "behat/mink": "1.6@stable",
        "behat/mink-extension": "*",
        "behat/mink-goutte-driver": "*",
        "behat/mink-selenium2-driver": "*",
        "behat/mink-browserkit-driver": "*",
        "drush/drush": "^9.0",
        "drupal/drupal-extension": "~3.0",
        "drupal/coder": "~8.2",
        "phpunit/phpunit": "~5.4",
        "squizlabs/php_codesniffer": "2.*",
        "phing/phing": "dev-master#0ef30e55bb5871cb38903cc0ee9d76074118a22c",
        "jakoch/phantomjs-installer": "1.9.8",
        "jarnaiz/behat-junit-formatter": "^1.2"
    },
    "autoload-dev": {
        "psr-4": {
            "Drupal\\Tests\\PHPUnit\\": "tests/phpunit/src/"
        }
    },
    "extra": {
        "installer-paths": {
            "docroot/core": [
                "type:drupal-core"
            ],
            "docroot/modules/contrib/{$name}": [
                "type:drupal-module"
            ],
            "docroot/profiles/contrib/{$name}": [
                "type:drupal-profile"
            ],
            "docroot/themes/contrib/{$name}": [
                "type:drupal-theme"
            ],
            "drush/contrib/{$name}": [
                "type:drupal-drush"
            ]
        },
        "patches": {
            "drupal/core": {
                "Ignore front end vendor folders to improve directory search performance": "https://www.drupal.org/files/issues/ignore_front_end_vendor-2329453-116.patch"
            }
        }
    },
    "scripts": {
        "install-phantomjs": "PhantomInstaller\\Installer::installPhantomJS",
        "post-install-cmd": [
            "PhantomInstaller\\Installer::installPhantomJS"
        ],
        "post-update-cmd": [
            "PhantomInstaller\\Installer::installPhantomJS"
        ]
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

Note that ```acquia/blt`` no longer appears in the dev requirements.  I believe that the template needs to include the dependency on BLT or else as soon as one runs any composer update commands, BLT is removed from the project.

**Pull request merged**